channel/player.go

```diff
@@ -1378,6 +1397,143 @@ func packetPlayerLevelUpAnimation(charID int32) mpacket.Packet {
 	return p
 }
 
+func packetPlayerSkillAnimSelf(charID int32, skillID int32, level byte) mpacket.Packet {
+	p := mpacket.CreateWithOpcode(opcode.SendChannelPlayerAnimation)
+	p.WriteInt32(charID)
+	p.WriteByte(0x01)
+	p.WriteInt32(skillID)
+	p.WriteByte(level)
+	return p
+}
+
+func packetPlayerSkillAnimThirdParty(charID int32, party bool, self bool, skillID int32, level byte) mpacket.Packet {
+	var p mpacket.Packet
+	if party && self {
+		p = mpacket.CreateWithOpcode(opcode.SendChannelSkillAnimation)
+	} else {
+		p = mpacket.CreateWithOpcode(opcode.SendChannelPlayerAnimation)
+		p.WriteInt32(charID)
+	}
+
+	if party {
+		p.WriteByte(0x02)
+	} else {
+		p.WriteByte(0x01)
+	}
+	p.WriteInt32(skillID)
+	// Basis uses WriteInt for level; encode as int32 to match
+	p.WriteInt32(int32(level))
+	p.WriteUint64(0)
+	p.WriteUint64(0)
+	return p
+}
+
+func packetPlayerGiveBuff(mask []byte, values []byte, delay int16, extra byte) mpacket.Packet {
+	p := mpacket.CreateWithOpcode(opcode.SendChannelTempStatChange)
+
+	// Normalize to 8 bytes (low dword, high dword)
+	if len(mask) < 8 {
+		tmp := make([]byte, 8)
+		copy(tmp[8-len(mask):], mask)
+		mask = tmp
+	} else if len(mask) > 8 {
+		mask = mask[len(mask)-8:]
+	}
+	p.WriteBytes(mask)
+
+	// Per-stat value triples (short value, int32 skill, short time)
+	p.WriteBytes(values)
+
+	// Self path: 2-byte delay
+	p.WriteInt16(delay)
+
+	// Optional extra (only if specific bits are present)
+
+	writeExtra := buffMaskNeedsExtraByte(mask)
+	if writeExtra {
+		p.WriteByte(extra)
+	}
+
+	p.WriteInt64(0)
+	p.WriteInt64(0)
+
+	return p
+}
+
+func packetPlayerGiveForeignBuff(charID int32, mask []byte, values []byte, extra byte) mpacket.Packet {
```

Comment:
Is this applying a buff to other players e.g. when applying haste this packet is sent for each party member sharing the buff?